### ISSUE TITLE
shellhook not being executed on cached environment

### ISSUE THREAD

**Issue by mogorman** [nix-community/nix-direnv#149]
```
direnv: using flake
direnv: renewed cache
SHELL HOOK FIRED
direnv: export +AR +AS +CC +CONFIG_SHELL +CXX +HEX_HOME +HOST_PATH +IN_NIX_SHELL +LD +MIX_HOME +NIX_BINTOOLS +NIX_BINTOOLS_WRAPPER_TARGET_HOST_x86_64_unknown_linux_gnu +NIX_BUILD_CORES +NIX_CC +NIX_CC_WRAPPER_TARGET_HOST_x86_64_unknown_linux_gnu +NIX_CFLAGS_COMPILE +NIX_ENFORCE_NO_NATIVE +NIX_HARDENING_ENABLE +NIX_INDENT_MAKE +NIX_LDFLAGS +NIX_STORE +NM +OBJCOPY +OBJDUMP +PGDATA +PGDIR +PGHOST +PGLOG +RANLIB +READELF +SIZE +SOURCE_DATE_EPOCH +STRINGS +STRIP +buildInputs +buildPhase +builder +configureFlags +depsBuildBuild +depsBuildBuildPropagated +depsBuildTarget +depsBuildTargetPropagated +depsHostHost +depsHostHostPropagated +depsTargetTarget +depsTargetTargetPropagated +doCheck +doInstallCheck +dontAddDisableDepTrack +name +nativeBuildInputs +out +outputs +patches +phases +propagatedBuildInputs +propagatedNativeBuildInputs +shell +shellHook +stdenv +strictDeps +system ~PATH ~XDG_DATA_DIRS
```
```
direnv: using flake
direnv: using cached dev shell
direnv: export +AR +AS +CC +CONFIG_SHELL +CXX +HEX_HOME +HOST_PATH +IN_NIX_SHELL +LD +MIX_HOME +NIX_BINTOOLS +NIX_BINTOOLS_WRAPPER_TARGET_HOST_x86_64_unknown_linux_gnu +NIX_BUILD_CORES +NIX_CC +NIX_CC_WRAPPER_TARGET_HOST_x86_64_unknown_linux_gnu +NIX_CFLAGS_COMPILE +NIX_ENFORCE_NO_NATIVE +NIX_HARDENING_ENABLE +NIX_INDENT_MAKE +NIX_LDFLAGS +NIX_STORE +NM +OBJCOPY +OBJDUMP +PGDATA +PGDIR +PGHOST +PGLOG +RANLIB +READELF +SIZE +SOURCE_DATE_EPOCH +STRINGS +STRIP +buildInputs +buildPhase +builder +configureFlags +depsBuildBuild +depsBuildBuildPropagated +depsBuildTarget +depsBuildTargetPropagated +depsHostHost +depsHostHostPropagated +depsTargetTarget +depsTargetTargetPropagated +doCheck +doInstallCheck +dontAddDisableDepTrack +name +nativeBuildInputs +out +outputs +patches +phases +propagatedBuildInputs +propagatedNativeBuildInputs +shell +shellHook +stdenv +strictDeps +system ~PATH ~XDG_DATA_DIRS
```

working around by eval "$shellHook" after my use flake

**Comment by bbenne10** [comment]
This looks like a clone of #109. The fact that it is re-discovered indicates that the solution is not quite discoverable enough (even though you seem to have found it with `eval "$shellHook"`).

If I understand it properly, the nix 2.4 PR that @Mic92 has open (but which is not complete) should resolve this. I'm going to leave this open until that PR is resolved so that we can track it. I don't know that we currently have the bandwidth to solve the problem in the meantime.

**Comment by Mic92** [comment]
So many things to work on :) I don't know when I get to this. If someone beats me with the implementation or an alternative approach I am all open for it.

**Comment by Mic92** [comment]
Fixed in https://github.com/nix-community/nix-direnv/pull/153 ?

**Comment by bbenne10** [comment]
This should absolutely be fixed in #153! I'm going to close this. Please test out the latest revision (documentation on how is in the main README) and open a new issue if you have problems with the newest.

I don't quite have the time right this moment, but will stage a release (probably to 2.0? Bumping minimum version of nix in use seems "major") very soon so that fetching from github is not the only way to test.
